From 3429b39a53587ef3ce8221cda368fc5ec0f641b7 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Wed, 28 Nov 2018 17:26:41 -0200
Subject: [PATCH] Revert "Revert "Revert "efi_loader: efi_allocate_pages is too
 restrictive"""

This reverts commit 14deb5e628dd365d63002ebf0d85d2ea9804bad2.

Revert until a proper memory map is set in place for dragonboard820c, in
order to avoid boot failures when loading artifacts from sdcard/network.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 lib/efi_loader/efi_memory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_memory.c b/lib/efi_loader/efi_memory.c
index 5bd4f4d7fc..1c0ceace5f 100644
--- a/lib/efi_loader/efi_memory.c
+++ b/lib/efi_loader/efi_memory.c
@@ -348,7 +348,7 @@ efi_status_t efi_allocate_pages(int type, int memory_type,
 	switch (type) {
 	case EFI_ALLOCATE_ANY_PAGES:
 		/* Any page */
-		addr = efi_find_free_memory(len, -1ULL);
+		addr = efi_find_free_memory(len, gd->start_addr_sp);
 		if (!addr) {
 			r = EFI_NOT_FOUND;
 			break;
-- 
2.17.1

